counsel.el

```diff
@@ -124,7 +124,9 @@ To skip the `executable-find' check, start the string with a space."
                      (car (split-string cmd)))))
       (or (and (stringp program)
                (not (string= program ""))
-               (executable-find program))
+               (if (<= 27 emacs-major-version)
+                   (executable-find program (file-remote-p default-directory))
+                 (executable-find program)))
```

Comment:
This way of writing this runtime check gives rise to the following warning in Emacs versions 24.5-26:

```
In counsel-require-program:
counsel.el:128:21:Warning: executable-find called with 2 arguments, but
    accepts only 1
```

One solution is simply to pacify the byte-compiler:

```el
(if (< emacs-major-version 27)
    (executable-find program)
  (declare-function executable-find "files"
                    (command &optional remote))
  (executable-find program (file-remote-p default-directory)))
```

But better yet would be to perform the check only once, at load time:

```el
(defalias 'counsel--executable-find
  ;; Gained optional argument in 27.1.
  (if (>= emacs-major-version 27)
      #'executable-find
    (lambda (command &optional _remote)
      (executable-find command)))
  "Compatibility shim for `executable-find'.")
```

And then write:

```el
(or (and (stringp program)
         (not (string= program ""))
         (counsel--executable-find program t))
    (user-error "Required program \"%s\" not found in your path" program))
```